java/test/org/openqa/selenium/edge/EdgeDriverFunctionalTest.java

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.edge;
+
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.chromium.ChromiumNetworkConditions;
+import org.openqa.selenium.chromium.HasCasting;
+import org.openqa.selenium.chromium.HasCdp;
+import org.openqa.selenium.chromium.HasNetworkConditions;
+import org.openqa.selenium.remote.Augmenter;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.testing.JUnit4TestBase;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class EdgeDriverFunctionalTest extends JUnit4TestBase {
+
+  @Test
+  public void shouldAllowRemoteWebDriverToAugmentHasNetworkConditions() throws MalformedURLException {
+    WebDriver driver = new RemoteWebDriver(new URL("http://localhost:4444/"), new EdgeOptions());
```

Comment:
No! This will never work! The tests won't start a server on port 4444: they pick a random port. Worse, because you're extending `JUnit4TestBase`, you already have a `driver` to work with.

Reply:
I skimmed through the other tests, and the same thing happens in them. You can't rely on a selenium server being run on a specific port. If you _really_ need to create a new webdriver instance (you don't, because the test already has one), then use `new WebDriverBuilder` from the test package.

Reply:
Obviously, assign the augmented browser to a variable, otherwise you'll break test isolation pretty handily.

Reply:
It's obvious that these test files aren't the ideal place for RemoteWebDriver class instances, but this is all patterned off of the test you wrote for me: https://github.com/SeleniumHQ/selenium/commit/0a5b2d8e747a2252fb4a2abada713a68a58ffe38#diff-6ff6428e62eafbf44f2cee9400a71f0a7f3276807d4b611000b7434c582752a3R523-R536

I think I'm doing the right thing in the places where I'm not using a RemoteWebDriver instance 
e.g., https://github.com/SeleniumHQ/selenium/pull/9856/files#diff-0bc642dab2d557261d4507422be800146f1ff78a1889f6f728e7eae08a76953fR117

I have tests for both "Augmented" and "with Builder" and I don't need them both, but I'm not sure the actual right way to do this.

Reply:
Discussed over a call. We can tag tests with `selenium-remote` and this will automatically create a test target that runs the test with the selenium server running.

Reply:
fixed for driver tests